plugins/modules/mso_schema_template_vrf_contract.py

```diff
@@ -0,0 +1,265 @@
+
+    schema_path = 'schemas/{id}'.format(**schema_obj)
+
+    # Get template
+    templates = [t.get('name') for t in schema_obj.get('templates')]
+    if template not in templates:
+        mso.fail_json(msg="Provided template '{0}' does not exist. Existing templates: {1}".format(template, ', '.join(templates)))
+    template_idx = templates.index(template)
+
+    # Get VRF
+    vrfs = [e.get('name') for e in schema_obj.get('templates')[template_idx]['vrfs']]
+    if vrf not in vrfs:
+        mso.fail_json(msg="Provided vrf '{vrf}' does not exist. Existing vrfs: {vrfs}".format(vrf=vrf, vrfs=', '.join(vrfs)))
+    vrf_idx = vrfs.index(vrf)
+    vrf_obj = schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]
+
+    if not vrf_obj.get('vzAnyEnabled'):
+        mso.fail_json(msg="vzAny attribute on vrf '{0}' is disabled.".format(vrf))
+
+    # Get Contract
+    if contract:
+        provider_contracts = [c.get('contractRef') for c in schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyProviderContracts']]
+        consumer_contracts = [c.get('contractRef') for c in schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyConsumerContracts']]
+        contract_ref = mso.contract_ref(**contract)
+        if contract_ref in provider_contracts and contract.get('type') == 'provider':
+            contract_idx = provider_contracts.index(contract_ref)
+            contract_path = '/templates/{0}/vrfs/{1}/vzAnyProviderContracts/{2}'.format(template, vrf, contract_idx)
+            mso.existing = schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyProviderContracts'][contract_idx]
+        if contract_ref in consumer_contracts and contract.get('type') == 'consumer':
+            contract_idx = consumer_contracts.index(contract_ref)
+            contract_path = '/templates/{0}/vrfs/{1}/vzAnyConsumerContracts/{2}'.format(template, vrf, contract_idx)
+            mso.existing = schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyConsumerContracts'][contract_idx]
+        if mso.existing.get('contractRef'):
+            mso.existing['contractRef'] = mso.dict_from_ref(mso.existing.get('contractRef'))
+            mso.existing['relationshipType'] = contract.get('type')
+
+    if state == 'query':
+        if not contract:
+            provider_contracts = [dict(contractRef=mso.dict_from_ref(c.get('contractRef')),
+                                  relationshipType='provider') for c in schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyProviderContracts']]
+            consumer_contracts = [dict(contractRef=mso.dict_from_ref(c.get('contractRef')),
+                                  relationshipType='consumer') for c in schema_obj.get('templates')[template_idx]['vrfs'][vrf_idx]['vzAnyConsumerContracts']]
+            mso.existing = provider_contracts + consumer_contracts
+        elif not mso.existing:
+            mso.fail_json(msg="Contract '{0}' not found".format(contract.get('name')))
+
```

Comment:
Can we add support for query all contract of a VRF if we don't pass the contract name?

Reply:
Forget about this comment, the other modules don't do this anyway.